pkgs/development/idris-modules/build-idris-package.nix

```diff
@@ -2,40 +2,56 @@
 #
 # args: Additional arguments to pass to mkDerivation. Generally should include at least
 #       name and src.
-{ stdenv, idris, gmp }: args: stdenv.mkDerivation ({
-  preHook = ''
-    # Library import path
-    export IDRIS_LIBRARY_PATH=$PWD/idris-libs
-    mkdir -p $IDRIS_LIBRARY_PATH
-
-    # Library install path
-    export IBCSUBDIR=$out/lib/${idris.name}
-    mkdir -p $IBCSUBDIR
-
-    addIdrisLibs () {
-      if [ -d $1/lib/${idris.name} ]; then
-        ln -sv $1/lib/${idris.name}/* $IDRIS_LIBRARY_PATH
-      fi
-    }
-
-    envHooks+=(addIdrisLibs)
+{ stdenv, idrisPackages, gmp }:
+  { idrisDeps ? []
+  , pkgName
+  , version
+  , src
+  , meta
+  , extraBuildInputs ? []
+  , postPatch ? ""
+  , postUnpack ? ""
+  , doCheck ? true
+  }:
+let
+  idris-with-packages = idrisPackages.with-packages (idrisDeps);
+in
+stdenv.mkDerivation ({
+
+  name = "${pkgName}-${version}";
+
+  postUnpack = postUnpack;
+
+
+  # Some packages use the style
+  # opts = -i ../../path/to/package
+  # rather than the declarative pkgs attribute so we have to rewrite the path.
+  postPatch = ''
+    sed -i *.ipkg -e "/^opts/ s|-i \\.\\./|-i ${idris-with-packages}/libs/|g"
+    cat *.ipkg
```

Comment:
Debug leftover?

Reply:
No, this is necessary to build the base packages.

Reply:
The `cat` is?